File: a/arc108_a.js
const I = line => Number.parseInt(line);
const LI = line => line.split(' ').map(v => Number.parseInt(v));
const AR = (n, ini) => Array(n).fill(ini);

const main = (arg) => {
    const inputs = arg.trim().split('\n');
    const [s, p] = LI(inputs[0]);
    let ans = 'No';
    for (let i = 1; i < (Number.parseInt(Math.sqrt(p))+1); i++) {
      if (i*(s-i) == p) ans = 'Yes';
    }
    console.log(ans);
  };
  main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: a/arc108_a.py
import sys
import math


def input(): return sys.stdin.readline().strip()
def I(): return int(input())
def LI(): return list(map(int, input().split()))
def IR(n): return [I() for i in range(n)]
def LIR(n): return [LI() for i in range(n)]
def SR(n): return [S() for i in range(n)]
def S(): return input()
def LS(): return input().split()


INF = float('inf')


s, p = LI()
for i in range(1, int(math.sqrt(p)) + 1):
    if i*(s-i) == p:
        print('Yes')
        exit()
print('No')
